pkg/controller/machine-set-boot-image/machine_set_boot_image_controller.go

```diff
@@ -198,11 +198,22 @@ func (ctrl *Controller) processNextWorkItem() bool {
 }
 
 func (ctrl *Controller) handleErr(err error, key interface{}) {
+
+	_, msName, splitErr := cache.SplitMetaNamespaceKey(key.(string))
+	if splitErr != nil {
+		klog.V(2).Infof("Exception %v handling error during machineset sync %v: %v", splitErr, key, err)
+		return
+	}
+
 	if err == nil {
 		ctrl.queue.Forget(key)
+		// Clear any errors as this machineset was successfully patched
+		ctrlcommon.MCCBootImageUpdateErr.WithLabelValues(msName).Set(0)
 		return
 	}
 
+	// Increment error on this machineset
+	ctrlcommon.MCCBootImageUpdateErr.WithLabelValues(msName).Inc()
```

Comment:
Do we want to count number of retry or just setting the value to 1 to indicate the failure?

Reply:
Hmm, I suppose both could work! I figured the number attempts would be more "live" information to the user when they get the alert . I'm open to either way - the alert should trigger if it is non zero (:

Reply:
No strong opinion, let's keep it the way it is and we can update later if needed.